plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2271,9 +2275,9 @@ private static boolean canUseInPredicate(ColumnMetadata column)
                 || type.equals(VARCHAR);
     }
 
-    private static Domain buildColumnDomain(ColumnMetadata column, DeltaLakeFileStatistics stats, List<String> canonicalPartitionColumns)
+    private static Domain buildColumnDomain(DeltaLakeColumnMetadata column, DeltaLakeFileStatistics stats, List<String> canonicalPartitionColumns)
     {
-        Optional<Long> nullCount = stats.getNullCount(column.getName());
+        Optional<Long> nullCount = stats.getNullCount(column.getPhysicalName());
```

Comment:
Do we have tests for SHOW STATS with 'name' mode?

Reply:
TestDeltaLakeColumnMappingMode has:
* testColumnMappingModeNameWithNonLowerCaseColumnName
* testShowStatsFromJsonForColumnMappingModeName
* testShowStatsFromParquetForColumnMappingModeName